client/components/date-picker/index.jsx

```diff
@@ -203,24 +215,39 @@ class DatePicker extends PureComponent {
 			);
 		}
 
+		if ( hasDateRange ) {
+			modifiers[ 'range-start' ] = this.props.selectedDays.from;
+			modifiers[ 'range-end' ] = this.props.selectedDays.to;
+			modifiers.range = {
+				from: this.props.selectedDays.from,
+				to: this.props.selectedDays.to,
+			};
+		}
+
+		const rootClassNames = classNames( {
+			'date-picker': true,
```

Comment:
A tip for classNames that are always true, we can pass a regular strings as arguments to `classNames` like so:

```js
const rootClassNames = classNames(
  'date-picker',
  `date-picker--${ this.props.numberOfMonths }up`,
  {
    somethingConditional: someCondition,
  }
);

(Not a blocker)